sqlmesh/core/engine_adapter/mssql.py

```diff
@@ -40,25 +42,53 @@ class MSSQLEngineAdapter(
 
     DIALECT: str = "tsql"
     FALSE_PREDICATE = exp.condition("1=2")
+    SUPPORTS_TUPLE_IN = False
+    TRAILING_SEMICOLON = True
 
-    def table_exists(self, table_name: TableName) -> bool:
-        """
-        Similar to Postgres, MsSql doesn't support describe so I'm using what
-        is used there and what the redshift cursor does to check if a table
-        exists. We don't use this directly in order for this to work as a base
-        class for other postgres.
+    def columns(
+        self,
+        table_name: TableName,
+        include_pseudo_columns: bool = True,
+    ) -> t.Dict[str, exp.DataType]:
+        """MsSql doesn't support describe so we query information_schema."""
 
-        Reference: https://github.com/aws/amazon-redshift-python-driver/blob/master/redshift_connector/cursor.py#L528-L553
-        """
         table = exp.to_table(table_name)
 
-        catalog_name = table.args.get("catalog") or "master"
+        catalog_name = table.catalog or "master"
```

Comment:
This logic is repeated to assign the default catalog name of `master`. One option would be to create a class variable at the top that is `DEFAULT_CATALOG_NAME` and set that to `master` and then replace this with `table.catalog or self.DEFAULT_CATALOG_NAME`. Another would be to create a method like `__get_catalog(table: Table) -> str: return table.cataog or "master"`. And then do `self.__get_catalog(table)` here.